backend/api/MQTT/MqttService.cs

```diff
@@ -115,25 +122,54 @@ private void OnConnected(MqttClientConnectedEventArgs obj)
                 _serverHost,
                 _serverPort
             );
+            _reconnectAttempts = 0;
         }
 
         private void OnConnectingFailed(ManagedProcessFailedEventArgs obj)
         {
+            string errorMsg =
+                "Failed to connect to MQTT broker. Exception: " + obj.Exception.Message;
+
+            if (_reconnectAttempts >= _maxRetryAttempts)
+            {
+                _logger.LogError("{errorMsg}\n      Exceeded max reconnect attempts.", errorMsg);
+
+                if (_shouldFailOnMaxRetries)
+                {
+                    _logger.LogError("Stopping MQTT client due to critical failure");
```

Comment:
Should we crash the entire backend or just the mqtt service if this is reached?

Reply:
If we crash the mqtt service, the backend also crashes.
We could configure ASP.NET to not crash on the crash of a Hosted Service, but the hosted service will never be restarted so that doesn't seem like a good idea.

Reply:
The backend still do not stop when the retries are used up.